content/en/docs/tasks/access-kubernetes-api/configure-aggregation-layer.md

```diff
@@ -22,13 +22,171 @@ Configuring the [aggregation layer](/docs/concepts/extend-kubernetes/api-extensi
+
+Note that the same client certificate is used by the Kubernetes apiserver to authenticate against _all_ extension apiservers. It does not create a client certificate per extension apiserver, but rather a single one to authenticate as the Kubernetes apiserver. This same one is reused for all extension apiserver requests. 
+
+#### Original Request Username and Group
+
+When the Kubernetes apiserver proxies the request to the extension apiserver, it informs the extension apiserver of the username and group with which the original request successfully authenticated. It provides these in http headers of its proxied request. You must inform the Kubernetes apiserver of the names of the headers to be used.
+
+* the header in which to store the username via `--requestheader-username-headers`
+* the header in which to store the group via `--requestheader-group-headers`
+* the prefix to append to all extra headers via `--requestheader-extra-headers-prefix`
+
+These header names are also placed in the `extension-apiserver-authentication` configmap, so they can be retrieved and used by extension apiservers.
+
+### Extension Apiserver Authenticates the Request
+
+The extension apiserver, upon receiving a proxied request from the Kubernetes apiserver, must validate that the request actually did come from a valid authenticating proxy, which role the Kubernetes apiserver is fulfilling. The extension apiserver validates it via:
+
+1. Retrieve the following from the configmap in `kube-system`, as described above:
+    * Client CA certificate
+    * List of allowed names (CNs)
+    * Header names for username, group and extra info
+2. Check that the TLS connection was authenticated using a client certificate which:
+    * Was signed by the CA whose certificate matches the retrieved CA certificate.
+    * Has a CN in the list of allowed CNs, unless the list is blank, in which case all CNs are allowed.
+    * Extract the username and group from the appropriate headers
+
+If the above passes, then the request is a valid proxied request from a legitimate authenticating proxy, in this case the Kubernetes apiserver.
+
+Note that it is the responsibility of the extension apiserver implementation to provide the above. Many do it by default, leveraging the `k8s.io/apiserver/` package. Others may provide options to override it using command-line options.
+
+In order to have permission to retrieve the configmap, an extension apiserver requires the appropriate role. There is a default role named `extension-apiserver-authentication-reader` in the `kube-system` namespace which can be assigned.
+
+### Extension Apiserver Authorizes the Request
+
+The extension apiserver now needs to validate that the user/group retrieved from the headers are authorized to execute the given request. It does so by sending a standard [SubjectAccessReview](/docs/reference/access-authn-authz/authorization/) request to the Kubernetes apiserver. 
```

Comment:
it would probably be useful to note that the extension apiserver must be given permission to submit these checks, and that there is a default clusterrole (system:auth-delegator) that allows submitting those checks that can be granted to the extension apiserver's service account

Reply:
Adding right now...